### PR TITLE
[BUGFIX] Missing loading of json_response extension classes

### PR DESCRIPTION
**Pull request by alexander-nitsche**
Resolves: #188
Releases: master

### CODE REVIEW

File: Classes/Core/Functional/Framework/Frontend/RequestBootstrap.php [hunk 1]
```diff
@@ -160,6 +161,7 @@ public function executeAndOutput()
             chdir($_SERVER['DOCUMENT_ROOT']);
             SystemEnvironmentBuilder::run(0, SystemEnvironmentBuilder::REQUESTTYPE_FE);
             $container = Bootstrap::init($this->classLoader);
+            ClassLoadingInformation::registerClassLoadingInformation();
```
Comment (bmack) [inline Classes/Core/Functional/Framework/Frontend/RequestBootstrap.php:164]: Maybe we could add some notes why we would add this as Bootstrap::init() calls the same method as well (if in non-composer mode)

Comment (alexander-nitsche) [reply Classes/Core/Functional/Framework/Frontend/RequestBootstrap.php:164]: It is aligned with the bootstrapping process of the FunctionalTestCase class [here](https://github.com/TYPO3/testing-framework/blob/master/Classes/Core/Testbase.php#L578) and [here](https://github.com/TYPO3/testing-framework/blob/master/Classes/Core/Testbase.php#L583). I would try to investigate why it was invented that way. Would you tell me, what you mean by non-composer setup: Is it the [Git + Composer Installation](https://get.typo3.org/version/10#accordion-git) (used for core development) or the [Wget / Curl / Download-Tar Installation](https://get.typo3.org/version/10#accordion-curl)?

Comment (alexander-nitsche) [reply Classes/Core/Functional/Framework/Frontend/RequestBootstrap.php]: > non-composer setup = Git + Composer Installation

I tested the additional class loading with the Git + Composer installation as well and it showed that it was not required for that type of TYPO3 installation as the `Bootstrap::init()` one line above already loads those additional classes. Thus i added the condition `Environment::isComposerMode()` in order to handle that and to be in line with the `Bootstrap::init()` class load handling. 

To summarize: 
Functional tests in _composer mode_ and in _non-composer mode_ load the additional classes in `\TYPO3\TestingFramework\Core\Testbase::setUpBasicTypo3Bootstrap` via 
```
$this->dumpClassLoadingInformation();
``` 
– as some lines above in 
```
$container = Bootstrap::init($classLoader);
``` 
no composer autoload files are available yet and have to be dumped first.
In a frontend request, the regarding response intercepting class `RequestBootstrap` has to reload those additional classes which is done in `\TYPO3\TestingFramework\Core\Functional\Framework\Frontend\RequestBootstrap::executeAndOutput()` – for _non-composer mode_ in 
```
$container = Bootstrap::init($this->classLoader);
```
and for _composer mode_ in the code lines of this PR.

I also moved this loading of classes into a new method `Testbase::reloadClassLoadingInformation` to be aligned with method `Testbase::dumpClassLoadingInformation`.

File: Classes/Core/Testbase.php [hunk 2]
```diff
@@ -598,6 +598,18 @@ public function dumpClassLoadingInformation()
...
     }
 
+    /**
+     * Reload class loading information if available
+     *
+     * @return void
+     */
+    public function reloadClassLoadingInformation()
+    {
+        if (ClassLoadingInformation::isClassLoadingInformationAvailable()) {
```
Comment (bmack) [inline Classes/Core/Testbase.php:608]: Hmm, this means, we reload ONLY if it exists, right? So it doesn't generate it "on the first run"?

Comment (bmack) [reply Classes/Core/Testbase.php:608]: We can also move this to the RequestBootstrap class directly.

Comment (alexander-nitsche) [reply Classes/Core/Testbase.php:608]: > Hmm, this means, we reload ONLY if it exists, right? So it doesn't generate it "on the first run"?

Yes, this function only loads classes which were dumped already. In RequestBootstrap we can be sure to have dumped classes already so we only need to load them again.

> We can also move this to the RequestBootstrap class directly.

Sure, i can move it back to RequestBootstrap directly.

Comment (alexander-nitsche) [reply Classes/Core/Testbase.php]: I moved it back to RequestBootstrap.

**Pull request merged**